client/src/main/java/org/asynchttpclient/netty/request/body/NettyBodyBody.java

```diff
@@ -77,10 +78,18 @@ public void onError(Throwable t) {
 
     channel.write(msg, channel.newProgressivePromise())
             .addListener(new WriteProgressListener(future, false, getContentLength()) {
+              @Override
               public void operationComplete(ChannelProgressiveFuture cf) {
                 closeSilently(body);
                 super.operationComplete(cf);
               }
+
+              @Override
+              protected boolean abortOnThrowable(Channel channel, Throwable cause) {
+                // FIXME dirty hack until netty issue is not resolved, see https://github.com/netty/netty/issues/6706
+                return StackTraceInspector.recoverOnChunkedUploadFailed(cause)
```

Comment:
Actually, do we really need stacktrace inspection? 
We already know that we're uploading a body. Is there really a way to get a `ChannelOutputShutdownException` whose stacktrace would pass though `ChunkedWriteHandler#flush` (note that there's still some work to do with FileRegion)?

Then, my concern is that I'm not sure wether a `ChannelOutputShutdownException` always means that we'll still be able to read nor if there's anything to read at all. I'm afraid if we don't have a response to read, we'll hang in half closed state forever.

Reply:
Yes, you're right, there is no need to check stack trace in this case.
Regarding FileRegion - it is not affected by this issue, since ChunkedWriteHandler handles it in totally different manner.